apps/src/code-studio/components/header/ProjectRemix.jsx

```diff
@@ -21,7 +21,7 @@ class ProjectRemix extends React.Component {
     ) {
       dashboard.project.serverSideRemix();
     } else if (!this.props.isSignedIn) {
-      window.location.assign(
+      utils.navigateToHref(
```

Comment:
We fix by switching to our own `navigateToHref` function, which exists entirely for making scenarios like this testable.

https://github.com/code-dot-org/code-dot-org/blob/209b008b3da5fa01ec1ecece15b283f9f6853081/apps/src/utils.js#L715-L723

And if you're wondering about the difference between `location.assign(x)` and `location.href = x`... [apparently there is none.](https://stackoverflow.com/questions/10302905/location-href-property-vs-location-assign-method)

Reply:
Note that `location.assign` enforces same-origin domain (see the specs for [`location.href =`](https://html.spec.whatwg.org/multipage/history.html#the-location-interface:dom-location-href-2) and [`location.assign`](https://html.spec.whatwg.org/multipage/history.html#dom-location-assign)); are we confident that's not relevant here?

Reply:
Hey, I had no idea.  Nice catch.  I'm pretty sure it's not relevant here because we've hard-coded `/users/sign_in` into the call, which is an origin-relative path to begin with.  The only thing I can think of is if there's some convoluted case with iframes, but `window.location != window.parent.location` so I think that's a non-issue too.